Management_System/DAL/Repositories/Implementations/TaskRepository.cs

```diff
@@ -0,0 +1,18 @@
+﻿using DAL.Context;
+using DAL.Entities;
+using DAL.Repositories.Interfaces;
+using Microsoft.EntityFrameworkCore;
+
+namespace DAL.Repositories.Implementations;
+
+public class TaskRepository(ManagementSystemDBContext db) : Repository<TaskEntity>(db), ITaskRepository
+{
+    public override async Task<TaskEntity?> GetById(Guid id, CancellationToken cancellationToken)
+    {
+        return await db.Set<TaskEntity>()
```

Comment:
You can use the Repository IQueryable method as base instead of Context Set() (if you have it, of course)